packages/react-utils/src/hooks/useSearchParams.ts

```diff
@@ -16,9 +16,14 @@ export function useSearchParams() {
     if (!value) {
       return;
     }
-    if (queryKey) {
-      sParams.delete(queryKey);
+
+    const keys = sParams.keys();
+    const keysArray = Array.from(keys);
+
+    if (keysArray.length > 2 || keysArray.length < 2) {
+      sParams.delete(keysArray[keysArray.length - 1]);
```

Comment:
this condition can be simplified to `(keysArray.length != 2)`, and It does not cover all cases, its actually wrong.

IMO, The approach where you update a key by first deleting it if exists, should be the preferred approach, simple and effective. Was there a reason you fore-went that for this?